### PR TITLE
Add license check script for incompatible licenses in pip dependencies

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
None

### Summary of the issue:
NVDA is licensed with GPL2 which is incompatible with certain licenses like Apache.
Currently there are 2 python dependencies bundled with NVDA with incompatible licenses:

- fast-diff-match-patch - see #16633 
- requests

### Description of user facing changes
- Developers can now check licenses with `runlicensecheck.bat`
- AppVeyor checks that new dependencies with incompatible licenses aren't introduced

### Description of development approach
- Using the [licensecheck](https://github.com/FHPythonUtils/LicenseCheck) pip dependency, check licenses
- Similar to lint checks, run these checks on appveyor builds

### Testing strategy:
 - [x] locally tested running licensecheck
 - [x] Test a succeeding appveyor build: https://ci.appveyor.com/project/NVAccess/nvda/builds/50104581
 - [x] Test a failing appveyor build: https://ci.appveyor.com/project/NVAccess/nvda/builds/50104676

### Known issues with pull request:
This doesn't check other dependencies such as those in `include` or `miscDeps`

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Introduced a license check process to automatically verify compatibility of new Python dependencies.

- **Documentation**
  - Updated contributing guidelines to include steps for running the new license check.
  - Expanded automated testing documentation to cover the new license check process.
  - Added build system notes about the new `runlicensecheck.bat` script.

- **Dependencies**
  - Added `licensecheck` package to `requirements.txt`.
  - Updated constraints and sources for dependencies like `Pillow`, `wxPython`, and `configobj`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -51,3 +51,30 @@ ignore = [
...
+	"robotframework",
+	"robotframework-screencaplibrary",
+	"robotremoteserver",
+
+	# Compatible licenses:
+	"certifi",  # Mozilla Public License 2.0
+	"markdown-link-attr-modifier",  # GPLV3 license, but not in PyPI correctly
+	"pycaw",  # MIT license, but not in PyPI
+	"wxPython",  # wxWindows Library License
+	"pillow",  # PIL Software License
```
Comment (seanbudd) [inline pyproject.toml:80]:
```suggestion
	"pillow",  # PIL Software License
]
```

File: projectDocs/testing/automated.md [hunk 2]
```diff
@@ -50,3 +50,10 @@ To run all tests standard tests for developers use `runsystemtests.bat --include
 Internally this script uses the Robot test framework to execute the tests.
 Any arguments given to `runsystemtests.bat` are forwarded onto Robot.
 For more details (including filtering and exclusion of tests) see `tests/system/readme.md`.
+
+### License checks
+
+NVDA uses GPLv2 which is incompatible with certain licenses like Apache.
```
Comment (gerald-hartig) [inline projectDocs/testing/automated.md:56]: is it "v2" or "v2 or later"

File: runlicensecheck.bat [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@echo off
+rem runlicensecheck [<output file>]
+rem Runs a license check for python dependencies
+set hereOrig=%~dp0
+set here=%hereOrig%
+if #%hereOrig:~-1%# == #\# set here=%hereOrig:~0,-1%
```
Comment (gerald-hartig) [inline runlicensecheck.bat:6]: Is this perhaps clearer in intent?
```suggestion
if #%here:~-1%# == #\# set here=%here:~0,-1%
```

**Review by gerald-hartig — APPROVED** [review]

File: projectDocs/testing/automated.md (hunk 2 above)
Comment (seanbudd) [reply projectDocs/testing/automated.md:56]: our current license is just GPLv2 with 2 modification
https://github.com/nvaccess/nvda/blob/8fb8ffcaba3950b1e4ca7b58e165267fc5369eb9/copying.txt

File: runlicensecheck.bat (hunk 3 above)
Comment (seanbudd) [reply runlicensecheck.bat:6]: I'm not sure, this is magic sauce at the top of each of our `.bat` files. I'm afraid to change it without updating them all, and concerned this suggestion is different in handling

**Pull request merged**